File: Leetcode/9. Palindrome Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0) return false;
        
        long div = 1;
        while(x >= 10*div) div *= 10;
        
        while(div >= 10) {
            int l = x%10, r = x/div;
            if(l != r) return 0;
            
            x %= div;
            x /= 10;
            
            div /= 100;
        }
        
        return 1;
    }
};

File: Leetcode/TypeScript/9. Palindrome Number.ts
function isPalindrome(x: number): boolean {
    if(x<0) return false;
    let str:string = x.toString();
    let i:number=0, j:number=str.length-1;
    while(i<j) {
        if(str[i] !== str[j]) return false;
        i++; j--;
    }
    return true;
};
